saleae/saleae.py

```diff
@@ -439,111 +447,132 @@ def load_from_file(self, file_path_on_target_machine):
+			if digital_channels is not None and len(digital_channels) and analog_channels is not None and len(analog_channels):
+				self._build('ANALOG_AND_DIGITAL')
+
+			# Add in the channels
 			if digital_channels is not None and len(digital_channels):
-				self._build('digital_channels')
-				for ch in digital_channels:
-					self._build(str(ch))
+				self._build(['{0:d} DIGITAL'.format(ch) for ch in digital_channels])
 			if analog_channels is not None and len(analog_channels):
-				self._build('analog_channels')
-				for ch in analog_channels:
-					self._build(str(ch))
-		if analog_channels is not None and len(analog_channels):
-			if analog_format not in ('voltage', 'adc'):
-				raise NotImplementedError("bad analog_format")
-			self._build(analog_format)
+				self._build(['{0:d} ANALOG'.format(ch) for ch in analog_channels])
+				is_analog = True
 
+		# Time selection
 		if time_span is None:
-			self._build('all_time')
+			self._build('ALL_TIME')
 		elif len(time_span) == 2:
-			self._build('time_span')
-			self._build(str(time_span[0]))
-			self._build(str(time_span[1]))
+			self._build(['TIME_SPAN', '{0:f}'.format(time_span[0]), '{0:f}'.format(time_span[0])])
 		else:
-			raise NotImplementedError('invalid time format')
-
-		if format == 'csv':
-			self._build(format)
+			raise ValueError('bad time span given')
 
-			if csv_column_headers:
-				self._build('headers')
-			else:
-				self._build('no_headers')
-
-			if csv_delimeter not in ('comma', 'tab'):
-				raise NotImplementedError('bad csv delimeter')
-			self._build(csv_delimeter)
-
-			if csv_timestamp not in ('time_stamp', 'sample_number'):
-				raise NotImplementedError('bad csv timestamp')
-			self._build(csv_timestamp)
-
-			if csv_combined:
-				self._build('combined')
-			else:
-				self._build('separate')
-
-			if csv_row_per_change:
-				self._build('row_per_change')
-			else:
-				self._build('row_per_sample')
-
-			if csv_number_format not in ('dec', 'hex', 'bin', 'ascii'):
-				raise NotImplementedError('bad csv number format')
-			self._build(csv_number_format)
-		elif format == 'bin':
-			self._build(format)
-
-			if bin_per_change:
-				self._build('on_change')
-			else:
-				self._build('each_sample')
-
-			if bin_word_size not in ('8', '16', '32', '64'):
-				raise NotImplementedError('bad bin word size')
-			self._build(bin_word_size)
+		# Find exporter
+		export_name = '_export_data_{0:s}_{1:s}'.format('analog' if is_analog else 'digital', format.lower())
+		if not hasattr(self, export_name):
+			raise NotImplementedError('bad export format given ({0:s})'.format(export_name))
 
-		elif format in ('vcd', 'matlab'):
-			# No options for these
-			self._build(format)
-		else:
-			raise NotImplementedError('unknown format')
+		# Let specific export function handle arguments
+		self._build(format.upper())
+		getattr(self, export_name)(**export_args)
 
+		# Done, return self for chainability
```

Comment:
Nothing else does this right now. I'd be willing to entertain the idea in a different PR, but let's not make an inconsistent API. Method chaining isn't very common in Python.